prospero-common/src/main/java/org/wildfly/prospero/test/MetadataTestUtils.java

```diff
@@ -0,0 +1,44 @@
+package org.wildfly.prospero.test;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.commons.io.FileUtils;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.wildfly.channel.Channel;
+import org.wildfly.channel.Stream;
+import org.wildfly.prospero.api.InstallationMetadata;
+import org.wildfly.prospero.api.exceptions.MetadataException;
+import org.wildfly.prospero.model.ChannelRef;
+
+public final class MetadataTestUtils {
+
+    private MetadataTestUtils() {
+    }
+
+    public static Channel createManifest(Collection<Stream> streams) {
+        return new Channel("manifest", null, null, null, streams);
+    }
+
+    public static InstallationMetadata createInstallationMetadata() throws IOException, MetadataException {
+        return createInstallationMetadata(createManifest(null), Collections.emptyList(), Collections.emptyList());
+    }
+
+    public static InstallationMetadata createInstallationMetadata(List<ChannelRef> channels,
+            List<RemoteRepository> repositories) throws IOException, MetadataException {
+        return createInstallationMetadata(createManifest(null), channels, repositories);
+    }
+
+    public static InstallationMetadata createInstallationMetadata(Channel manifest, List<ChannelRef> channels,
+            List<RemoteRepository> repositories) throws MetadataException, IOException {
+        Path installation = Files.createTempDirectory("installation");
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> FileUtils.deleteQuietly(installation.toFile())));
```

Comment:
How about using TemporaryFolder junit Rule? It could be a constructor argument and we don't need to worry about cleanup

Reply:
Good idea.